UPGRADING.md

```diff
@@ -1,6 +1,16 @@
 Upgrading SlackRubyBot
 ======================
 
+### Upgrading to >= 0.8.0
+
+#### Hook::Base was removed, explicitly register any custom hook classes you have
+
+Hook classes are now handled differently, namely they are explicitly registered into `SlackRubyBot::Server` via a configuration option (`hook_handlers`) or by passing a similar hash later on through the `add_hook_handlers` method. Including Hook classes directly into the server class is no longer needed.
+
+A hook is actioned via a `call` message onto the handler object (this can be anything that responds to that), so you'll need to rename your method names.
+
+Finally, you can now register multiple hooks for the same event, so if you had any code to remove default hooks, you'll need to change it so you pass a configuration hash into `Server`
```

Comment:
Not sure if there was a canonical approach, you could maybe manipulate the `hooks` attribute, but I'm not sure people would do that. That was actually what triggered this (https://github.com/dblock/slack-ruby-bot/issues/50) .

Would that be a fair example you think?